### PR TITLE
Add webUI tests for share only with membership groups

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Add webUI tests for share with only membership groups.

## Related Issue
#35673

## How Has This Been Tested?
🤖

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 1]
```diff
@@ -276,6 +279,22 @@ public function enableExcludeGroupFromSharing(Session $session) {
...
 	}
 
+	/**
+	 * enable restrict users to only share with groups they are member of
+	 *
+	 * @param Session $session
+	 *
+	 * @return void
+	 */
+	public function restrictUserToOnlyShareWithMembershipGroup(Session $session) {
```
Comment (dpakach) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: I think you have to check if the checkbox is already cheked.

Comment (paurakhsharma) [reply tests/acceptance/features/lib/AdminSharingSettingsPage.php]: `$this->toggleCheckbox()` does that automatically for you ;)

Comment (dpakach) [reply tests/acceptance/features/lib/AdminSharingSettingsPage.php]: oh, ok got it.

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 2]
```diff
@@ -254,4 +254,42 @@ Feature: Sharing files and folders with internal groups
...
+    And the user types "grp2" in the share-with-field
+    Then a tooltip with the text "No users or groups found for grp2" should be shown near the share-with-field on the webUI
+
+  Scenario: sharer should be able to share a folder to his/her own group when only share with groups they are member of is enabled
+    Given group "grp2" has been created
+    And user "user1" has created folder "/simple-folder"
+    And the administrator has browsed to the admin sharing settings page
+    When the administrator enables restrict users to only share with groups they are member of using the webUI
+    And the user re-logs in as "user1" using the webUI
+    And the user shares folder "simple-folder" with user "user2" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: This is sharing with a user, but the scenario description says it should be sharing with his/her own group.

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 3]
```diff
@@ -254,4 +254,42 @@ Feature: Sharing files and folders with internal groups
...
+    Then as "user2" folder "/simple-folder" should exist
+
+  Scenario: sharer should be able to share a file to his/her own group when only share with groups they are member of is enabled
+    Given group "grp2" has been created
+    And user "user1" has uploaded file with content "some content" to "lorem.txt"
+    And the administrator has browsed to the admin sharing settings page
+    When the administrator enables restrict users to only share with groups they are member of using the webUI
+    And the user re-logs in as "user1" using the webUI
+    And the user shares file "lorem.txt" with user "user2" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: This is sharing with a user, but the scenario description says it should be sharing with his/her own group.

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature (hunk 2 above)
Comment (paurakhsharma) [reply tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: Agh, I must have copied the scenario from `shareWithUsers.feature` and only changed the scenario description but not steps somehow.

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on: #35753

**Pull request merged**